src/plugins/kibana_react/public/util/test_helpers/react_mount_serializer.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// eslint-disable-next-line import/no-default-export
+export default {
+  test(value: any) {
+    return value && value.__reactMount__;
+  },
+  print(value: any, serialize: any, indent: any) {
+    return `reactMount(${serialize(value.__reactMount__)})`;
```

Comment:
Most snapshots use proper names for the functions rather than calling points, and it would be *very* nice if the formatting correctly handled indentation. Maybe something like the following (untested)?

```ts
return `[MountPoint] {\n  ${serialize(value.__reactMount__)}\n}`;
```

Hopefully that should give us a consistent snapshot:

```ts
expect((notifications.toasts.add.mock.calls[0][0] as any).text).toMatchInlineSnapshot(`
  [MountPoint] {
    <React.Fragment>
      <div>
        baz
      </div>
    </React.Fragment>
  }
`);

expect((notifications.toasts.add.mock.calls[0][0] as any).text).toMatchInlineSnapshot(`
  [MountPoint] {
    <React.Fragment />
  }
`);
```

Reply:
`[MountPoint]` is better than `reactMount(` indeed, thanks.

However for the second part, I had issue with the indent when using manual \n:

`[MountPoint] {\n  ${serialize(value.__reactMount__)}\n}`;

breaks the indent and renders as something like
```ts
expect((notifications.toasts.add.mock.calls[0][0] as any).text).toMatchInlineSnapshot(`
  [MountPoint] {
<React.Fragment>
    <div>
      baz
    </div>
  </React.Fragment>
  }
`);
```

Even with trying things with the 3rd `indent` param that is passed to the `print`, I had nothing correct. will try again (very poor documentation is available on these APIs)

Reply:
Now rendering to something like
```jsx
"title": MountPoint {
      "reactNode": <FormattedMessage
        defaultMessage="Report may contain formulas {reportObjectType} '{reportObjectTitle}'"
        id="xpack.reporting.publicNotifier.csvContainsFormulas.formulaReportTitle"
        values={
          Object {
            "reportObjectTitle": "Yas",
            "reportObjectType": "yas",
          }
        }
      />,
    },
```

Had to trick the serializer for this, as the API is very limited, but it's definitely better.

Also reasoning behind having an actual `reactNode` property is that we may have other `MountPoint` adapters in the futur. having proper properties for each adapter will greatly help in snapshot testing